account_multicurrency_revaluation_type/tests/test_currency_revaluation.py

```diff
@@ -0,0 +1,173 @@
+            'invoice_line_ids': [(0, 0, invoice_line_data)],
+            'payment_term_id': payment_term.id,
+        })
+        # Validate invoice
+        invoice.action_invoice_open()
+
+        payment_method = ref('account_multicurrency_revaluation.'
+                             'account_payment_method_manual_in')
+
+        # Register partial payment
+        payment = cls.env['account.payment'].create({
+            'invoice_ids': [(4, invoice.id, 0)],
+            'amount': 500,
+            'currency_id': usd_currency.id,
+            'company_id': cls.company.id,
+            'payment_date': cls.jan_31,
+            'communication': 'Invoice partial payment',
+            'partner_id': invoice.partner_id.id,
+            'partner_type': 'customer',
+            'journal_id': bank_journal_usd.id,
+            'payment_type': 'inbound',
+            'payment_method_id': payment_method.id,
+            'payment_difference_handling': 'open',
+            'writeoff_account_id': False,
+        })
+        payment.post()
+
+    def test_revaluation_loss(self):
+        # We test that wizard take right currency rate - month rate instead
+        # of daily rate
+
+        # re-estimate date 01-31 rate 5.40
+        # lines selected for revaluation
+        # amount_currency(USD)  balance(EUR)   account_id
+        #    500                 250           Account Liquidity USD
+        #    300                 150           Account Receivable
+        # lines estimated on rate 2.0 get from  currency_rate_usd_01
+        #
+        # in result we get 2 evaluation lines
+        # 300/5.4 = 55.56 - 150 = -94.44
+        # 500/5.4 = 92.59 - 250 = -157.41
+
+        result = self.wizard_execute(self.dec_31, 'average')
+        move_ids = self._get_move_ids(result)
+        debit = move_ids.filtered(lambda s: s.debit > 0)
+        self.assertEqual(sum(debit.mapped('balance')), 251.85)
+
+    def test_revaluation_gain(self):
+        # date 04-05 rate 1.20
+        # lines selected for revaluation same
+        # in result we get 2 evaluation lines
+        # 300/1.2 = 250 - 150 = 100
+        # 500/1.2 = 416.67 - 250 = 166.67
+
+        result = self.wizard_execute(self.march_04, 'average')
+        move_ids = self._get_move_ids(result)
+        debit = move_ids.filtered(lambda s: s.credit > 0)
+        self.assertAlmostEqual(sum(debit.mapped('balance')), -266.67)
+
+    def _get_move_ids(self, result):
+        move_ids = []
+        lines = result['domain'].split(',')
+        for l in lines:
+            dig = re.search(r'\d+', l)
+            if dig:
+                move_ids.append(int(dig.group()))
+        move_ids = self.env['account.move.line'].browse(move_ids)
+        return move_ids
+
+    def wizard_execute(self, date, type):
+
+        data = {
+            'revaluation_date': date,
+            'journal_id': self.reval_journal.id,
+            'label': '[%(account)s] [%(currency)s] wiz_test',
```

Comment:
Would be great if you test the rate which is displayed in the label to make sure it's the monthly one. cf https://github.com/OCA/account-closing/pull/103/commits/ecb4ebe45cff9efa525c4ed213ab894759e7776e

Reply:
Can you please add a test for that ? Will fix mine on 12.0 asap :laughing: